enhancements/security/runtime-class-scc.md

```diff
@@ -0,0 +1,290 @@
+while also providing protection from less privileged users gaining access to features they should not.
+
+### Goals
+
+- SCC should support specifying the runtime classes that must be specified in the pod request from a user.
+
+### Non-Goals
+
+- Changing the current SCC profiles to allow any extra runtime classes.
+- Changing behavior of the default runtime class.
+- Adding additional SCCs for various customzied behavior using AllowedRuntimeClasses.
+
+## Proposal
+
+
+[RuntimeClasses][runtime-classes] is a feature in Kubernetes that allows a user to request a certain runtime configuration.
+CRI-O has enhanced this feature to not only branch on different runtimes, but customize the way the pod is run.
+With the customization of the way a pod is run comes risk of users allocating resources they should not be allowed to.
+This leads to the need for a way to gate different users from using different runtime classes,
+and SCC is a perfect component to do so.
+
+The proposal is to add a new field to the SCC: `RequiredRuntimeClasses`:
+
+The API for `RequiredRuntimeClasses` will follow the implementation of [RequiredDropCapabilities][required-drop-capabilities],
+where a specific SCC profile will be able to specify a set of runtime classes,
+and pods in that SCC must request a runtime class that matches that set.
+If a request for a runtime class comes in that doesn't match the allowed runtime classes, the Pod will be denied.
+
+However, instead of a customized type,
+or using a type from the kubernetes API like [the Runtime Class structure][runtime-class-structure],
+`AllowedRuntimeClasses` will be a slice of strings.
+One can think of it as a list of [Handlers][handlers] of the runtime classes that are allowed.
+
+```
+type SecurityContextConstraints struct {
+    ...
+	// RequiredRuntimeClasses is a list of names of runtime classes (also called runtime handlers).
+	// The runtime class specified in a pod request must match one of the items in this list.
+	// To ensure any runtime class, '*' may be specified.
+	// To ensure the default runtime class is used, an empty slice may be specified.
+	// +nullable
+	EnsuredRuntimeClasses []string `json:"ensuredRuntimeClasses" protobuf:"bytes,26,rep,name=ensuredRuntimeClasses"`
+    ...
+}
+```
+Fields in this slice will be logically OR'd.
+A user must specify an item that only matches one field to be permitted to use that runtime class.
+This allows an admin to specify a set of runtime classes that a user is allowed to use.
+
+This gives the admin three options for configuration:
+1. Specifying the empty slice (`[]`) requires pods in the SCC to not request a custom runtime class (meaning it is given the default).
+2. Specifying a set of items (`["openshift-builder"]`) requires pods in the SCC must request a runtime class that matches an item in the list.
+3. Specifying `["*"]` means pods int he SCC can request any runtime class, or omit requesting a custom one.
```

Comment:
TODO: maybe a special string that means "only the default runtime class is allowed"?